src/callbacks_step/time_series.jl

```diff
@@ -0,0 +1,200 @@
+  end
+
+  if ndims(point_coordinates) != 2 || size(point_coordinates, 2) != ndims(mesh)
+    throw(ArgumentError("`point_coordinates` must be a matrix of size n_points × ndims"))
+  end
+
+  # Transpose point_coordinates to our usual format [ndims, n_points]
+  # Note: They are accepted in a different format to allow direct input from `readdlm`
+  point_coordinates_ = permutedims(point_coordinates)
+
+  # Invoke callback every `interval` time steps or after final step (for storing the data on disk)
+  if interval > 0
+    condition = (u, t, integrator) -> (integrator.iter % interval == 0 || isfinished(integrator))
+  else # disable the callback for interval == 0
+    condition = (u, t, integrator) -> false
+  end
+
+  # Create data structures that are to be filled by the callback
+  variable_names = varnames(solution_variables, equations)
+  n_points = size(point_coordinates_, 2)
+  point_data = Vector{uEltype}[Vector{uEltype}() for _ in 1:n_points]
+  time = Vector{RealT}()
+  step = Vector{Int}()
+  time_series_cache = create_cache_time_series(point_coordinates_, mesh, solver, cache)
+
+  time_series_callback = TimeSeriesCallback(interval,
+                                           solution_variables,
+                                           variable_names,
+                                           output_directory,
+                                           filename,
+                                           point_coordinates_,
+                                           point_data,
+                                           time,
+                                           step,
+                                           time_series_cache)
+
+  return DiscreteCallback(condition, time_series_callback, save_positions=(false,false))
+end
+
+
+# Convenience constructor that unpacks the semidiscretization into mesh, equations, solver, cache
+function TimeSeriesCallback(semi, point_coordinates; kwargs...)
+  mesh, equations, solver, cache = mesh_equations_solver_cache(semi)
+
+  return TimeSeriesCallback(mesh, equations, solver, cache, point_coordinates; kwargs...)
+end
+
+
+# Convenience constructor that converts a vector of points into a Trixi-style coordinate array
+function TimeSeriesCallback(mesh, equations, solver, cache, point_coordinates::AbstractVector;
+                            kwargs...)
+  # Coordinates are usually stored in [ndims, n_points], but here as [n_points, ndims] 
+  n_points = length(point_coordinates)
+  point_coordinates_ = Matrix{eltype(eltype(point_coordinates))}(undef, n_points, ndims(mesh))
+
+  for p in 1:n_points
+    for d in 1:ndims(mesh)
+      point_coordinates_[p, d] = point_coordinates[p][d]
+    end
+  end
+
+  return TimeSeriesCallback(mesh, equations, solver, cache, point_coordinates_; kwargs...)
+end
+
+
+# This method is called as callback during the time integration.
+@inline function (time_series_callback::TimeSeriesCallback)(integrator)
```

Comment:
```suggestion
function (time_series_callback::TimeSeriesCallback)(integrator)
```
Do we need that here?

Reply:
I guess not. Copy paste 🤷